modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridDhtAtomicCache.java

```diff
@@ -1861,7 +1864,11 @@ private void updateAllAsyncInternal0(
                                                 + req.topologyVersion() + ']';
                                         }
                                         else {
-                                            affFut.listen(f -> updateAllAsyncInternal0(node, req, completionCb));
+                                            affFut.listen(f -> updateAllAsyncInternal0(
+                                                node,
+                                                req,
+                                                completionCb,
+                                                MTC.span()));
```

Comment:
Shouldn't it be span instead of MTC.span ?
updateAllAsyncInternal0 may postpone its own execution several times in a row

Reply:
Yep, it should be span. Fixed.